packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2415,6 +2463,8 @@ class Navigator extends StatefulWidget {
   /// the initial route.
   ///
   /// If there is no [Navigator] in scope, returns false.
+  ///
+  /// Does not consider [PopEntry].
```

Comment:
This is surprising, shouldn't it consider popEntry?

Reply:
Before this PR, it didn't consider WillPopScope, so I made it not consider PopEntry/PopScope either.  Fundamentally we do need a way to consider these two things separately (does the Navigator have a poppable route, and is the Navigator being prevented from popping by PopEntry) because PopScope.canPop is often set based on whether or not a Navigator has a poppable route.

I was trying to make this a bit more clear with docs like this, but maybe I need to expand a little bit more.